aws/logs_monitoring/lambda_function.py

```diff
@@ -805,15 +793,39 @@ def forward_metrics(metrics):
                 log.debug(f"Forwarded metric: {json.dumps(metric)}")
 
 
-def forward_traces(traces):
-    for trace in traces:
+def forward_traces(trace_payloads):
+    batched_payloads = batch_trace_payloads(trace_payloads)
+
+    for payload in batched_payloads:
         try:
-            trace_connection.send_trace(trace["message"], trace["tags"])
+            trace_connection.send_traces(payload["message"], payload["tags"])
         except Exception:
-            log.exception(f"Exception while forwarding trace {json.dumps(trace)}")
+            log.exception(f"Exception while forwarding traces {json.dumps(payload)}")
         else:
             if log.isEnabledFor(logging.DEBUG):
-                log.debug(f"Forwarded trace: {json.dumps(trace)}")
+                log.debug(f"Forwarded traces: {json.dumps(payload)}")
+
+
+def batch_trace_payloads(trace_payloads):
+    """
+    To reduce the number of API calls, batch traces that have the same tags
+    """
+    traces_grouped_by_tags = defaultdict(list)
+    for trace_payload in trace_payloads:
+        tags = trace_payload["tags"]
+        traces = json.loads(trace_payload["message"])["traces"]
+        traces_grouped_by_tags[tags] += traces
+
+    batched_trace_payloads = []
+    batcher = DatadogBatcher(256 * 1000, 2 * 1000 * 1000, 200)
```

Comment:
What are these constants?

Reply:
They are:
* The max allowed item size in bytes (256 KB)
* The max allowed batch size in bytes (2 MB)
* The max number of items per batch (200)

I copy-pasted these specific numbers from the other usage of DatadogBatcher in this file, which is for batching logs.

Reply:
Could we refactor them into constant variables?